ingester/src/deferred_load.rs

```diff
@@ -68,6 +73,85 @@ where
     }
 }
 
+impl<T> Display for DeferredLoad<T>
+where
+    T: Display,
+{
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self.value.lock().as_ref().unwrap() {
+            State::Unresolved(_) | State::Loading(_) => f.write_str(UNRESOLVED_DISPLAY_STRING),
+            State::Resolved(v) => v.fmt(f),
+        }
+    }
+}
+
+impl<T> DeferredLoad<T> {
+    /// Provide a hint to the [`DeferredLoad`] that the value will be used soon.
+    ///
+    /// This allows the value to be materialised in the background, in parallel
+    /// while the caller is executing code that will eventually call
+    /// [`Self::get()`].
+    pub(crate) fn prefetch_now(&self) {
+        let mut state = self.value.lock();
+
+        // If the value has already resolved, this call is a NOP.
+        if let Some(State::Resolved(_)) = &*state {
+            return;
+        }
+
+        // Potentially transition the state, discarding the waker.
+        let (_, new_state) = self.get_load_waker(state.take().unwrap());
+        *state = Some(new_state);
+    }
+
+    /// Potentially transition `state`, returning the new state and the
+    /// [`Notify`] that will be signalled when loading the value completes.
+    ///
+    /// # Panics
+    ///
+    /// This method panics if `state` is [`State::Resolved`].
+    fn get_load_waker(&self, state: State<T>) -> (Arc<Notify>, State<T>) {
```

Comment:
Again not a huge deal but potentially saving confusion in the future: The comment says this returns "new state and notify" and the return type has those in the opposite order. Could we change this to:

```suggestion
    /// Potentially transition `state`, returning the [`Notify`] that will be signalled when loading the value 
    /// completes and the new state.
    ///
    /// # Panics
    ///
    /// This method panics if `state` is [`State::Resolved`].
    fn get_load_waker(&self, state: State<T>) -> (Arc<Notify>, State<T>) {
```